internal/prompt/prompt.go

```diff
@@ -68,14 +70,25 @@ func (p *Prompt) Input(title, message string, defaultResponse *string, flags ...
 	}, flags...)
 }
 
+// interactiveInputError returns the proper input error for a non-interactive prompt.
+// If the terminal cannot show prompts (e.g. Git Bash on Windows), the error mentions this.
+// Otherwise, the error simply states the prompt cannot be resolved in non-interactive mode.
+// The "message" argument is the prompt's user-facing message.
+func interactiveInputError(message string) error {
+	if !term.IsTerminal(int(os.Stdin.Fd())) {
```

Comment:
What's the distinction between a non-interactive prompt and a non-interactive terminal? This seems like it might be worth passing along like we do non-interactive, rather than resolving it through use of a global.

Reply:
Non-interactive prompt is when you pass `-n` and do not want any interaction at all. Non-interactive terminal is when you don't pass `-n` because you wan't prompts, but cannot receive prompts because your terminal is non-interactive.

Reply:
I don't think that distinction is accurate. Yes the user can supply `-n` which would set non-interactive, but we also detect it:

https://github.com/ActiveState/cli/blob/4b38cc68f36736428be2ad5e42a94c9e431b414f/cmd/state/main.go#L98-L98

So this check should ideally rely on https://github.com/ActiveState/cli/blob/4b38cc68f36736428be2ad5e42a94c9e431b414f/internal/prompt/prompt.go#L40-L40

Reply:
The `terminal.IsTerminal()` call is part of a multi-line conditional statement. One line above is a check for `-n` or `--non-interactive`. If that option is present, the entire `isInteractive` bool becomes false, regardless of whether or not the terminal is actuall interactive. This is why the separate check exists in the prompt package.

Reply:
I'm not understanding how that's an issue. The user specifying it as non-interactive ought to be respected. It shouldn't encounter the error you're trying to raise in that condition anyway.

Reply:
I respected existing behavior with regards to the user specifying or not specifying non-interactive. The only thing I added was the case where the State Tool is prompting for input, but the user's terminal does not allow it. In this special case, we give a more informative error message. Otherwise, it's business as usual.

Reply:
(I clicked Delete instead of Edit *sigh*)
I am respecting the existing behavior regarding the user specifying interactive vs non-interactive. The only special case I'm handling is when the State Tool wants to prompt the user, but the terminal doesn't allow it. In that case I'm giving a more informative error message. Otherwise, it's business as usual.

Reply:
Removed IsTerminal() test in favor of existing `isInteractive`. Still use a separate message for Windows as opposed to macOS and Linux, but prefer "interactive mode" verbage as opposed to "interactive terminal".